tests/README.md

```diff
@@ -1,24 +1,40 @@
 # Integration tests
 
+
+### Run the tests
 To start the tests, run `python3 run_integration_tests.py` after starting a docker
-(`docker-compose up` at the root of the project)
+(`docker-compose up` at the root of the project).
+
+To run a list of specific tests, run `python3 run_integration_tests.py test_name_1 test_name_2 ...`.
+
+### Create new integration tests
 
 To add a test, create a python file in the `tests/` folder with the other tests.
 Inside, create a `run(*args, **kwargs)` function. It will be given some parameters
 in the `kwargs`, for now it contains:
 
 ```json
 {
-  "infra_id": new infra id (int),
+  "all_infras": {
+    "dummy": dummy infra id (int),
+    "tiny": tiny infra id (int),
+    "small": small infra id (int),
+  },
   "url": api url
 }
 ```
 
+The python file can instead contain a function `list_tests() -> Iterable[Tuple[string, Callable]]`.
+In this case, it will be called to get a list of tests.
+
 
 # Fuzzer
 
 The fuzzer is a script that generates random tests and logs any error that occurred.
-Run `python3 fuzzer/fuzzer.py`, any error will be logged in `fuzzer/errors`.
+Run `python3 fuzzer/fuzzer.py`, any error will be reported in `fuzzer/errors` in a json.
+
+Note: you need a docker running locally *with at least one infra imported*.
+It can be a generated infra, or it can be imported from some other DB.
 
-Note: you need a docker running locally *with a database setup with a real infra*.
-This data isn't publicly available yet. 
+If the test is run on a generated infra, the json containing the error report
+can be copied to `tests/fuzzer_outputs/` to integrate it into the test suite.
```

Comment:
I'm a bit confused, couldn't these be directly added to the backend test suite?
Usually, reproducing tests of former bugs are called regression tests. fuzzer_outputs is useful to know where these tests are from, yet not very useful to know what their intended purpose is.

Reply:
> I'm a bit confused, couldn't these be directly added to the backend test suite?

It's mort effort to adapt those to the backend test suite, one fuzzer test includes several requests to the back

> fuzzer_outputs is useful to know where these tests are from, yet not very useful to know what their intended purpose is.

It's named like that because the test suite expects the folder to contains a specific json format, generated from the fuzzer

Reply:
I like regression_fuzzer_tests better, I agree to keep it apart from the backend